evaluator/builtin_other.go

```diff
@@ -25,6 +26,28 @@ import (
 	"github.com/pingcap/tidb/util/types"
 )
 
+// See http://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_sleep
+func builtinSleep(args []types.Datum, _ context.Context) (d types.Datum, err error) {
+	if args[0].IsNull() {
```

Comment:
In the MySQL 5.7 doc has the following description: If the argument is NULL or negative, SLEEP() produces a warning, or an error in strict SQL mode.
I'll add SQL mode judgement.